packages/langium/src/grammar/lsp/grammar-code-actions.ts

```diff
@@ -66,6 +69,52 @@ export class LangiumGrammarCodeActionProvider implements CodeActionProvider {
         return undefined;
     }
 
+    /**
+    * Returns whether an AST node of a given type exists within the range of a diagnostic
+    * Helpful to prevent code actions being suggested in the incorrect context
+    *
+    * @param diagnostic To use to determine the range to search for a node
+    * @param document To search within
+    * @param astTypePredicate To select the desired node
+    * @returns Whether or not the node exists
+    */
+    private astNodeExistsAtDiagnostic<T extends AstNode>(diagnostic: Diagnostic, document: LangiumDocument, astTypePredicate: (n: AstNode) => n is T): boolean {
```

Comment:
Question: Does this method accomplish something except for catching errors from the validator? I believe the method only returns `false` if the validator returns incorrect diagnostic ranges, right?

Reply:
Yep, it was intended as a sanity check during development, where I had accidentally caught other nodes in the diagnostic. That was some months ago now, and looking back at it now...it can be dropped 🗑️

Reply:
Removed this